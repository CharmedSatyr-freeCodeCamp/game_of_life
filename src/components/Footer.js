import React from 'react';

const Footer = () => (
  <footer>
    <p>
      <a
        href="https://github.com/CharmedSatyr-freeCodeCamp/game_of_life"
        rel="noopener noreferrer"
        target="_blank"
      >
        Source code
      </a>{' '}
      available under the{' '}
      <a
        href="https://github.com/CharmedSatyr-freeCodeCamp/game_of_life/blob/master/LICENSE"
        rel="noopener noreferrer"
        target="_blank"
      >
        GNU Public License v3.0
      </a>
    </p>
  </footer>
);

export default Footer;
