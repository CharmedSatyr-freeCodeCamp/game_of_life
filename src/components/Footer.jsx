import React from 'react';

const Footer = () => (
    <footer>
        I 'm a footer.
    </footer>
)

export default Footer;
