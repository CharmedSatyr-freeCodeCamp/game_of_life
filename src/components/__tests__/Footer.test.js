import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Footer from '../Footer';

configure({ adapter: new Adapter() });

describe('<Footer /> component', () => {
  it('renders without crashing', () => {
    const footer = shallow(<Footer />);
    expect(footer.exists()).toBeTruthy();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
