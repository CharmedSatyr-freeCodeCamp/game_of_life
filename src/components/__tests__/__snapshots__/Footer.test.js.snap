// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> component renders correctly 1`] = `
<footer>
  <p>
    <a
      href="https://github.com/CharmedSatyr/game_of_life"
      rel="noopener noreferrer"
      target="_blank"
    >
      Source code
    </a>
     
    available under the
     
    <a
      href="https://github.com/CharmedSatyr/game_of_life/blob/master/LICENSE"
      rel="noopener noreferrer"
      target="_blank"
    >
      GNU Public License v3.0
    </a>
  </p>
</footer>
`;
